numpy/lib/tests/test_stride_tricks.py

```diff
@@ -415,12 +415,27 @@ def test_writeable():
     assert_equal(result.flags.writeable, False)
     assert_raises(ValueError, result.__setitem__, slice(None), 0)
 
-    # but the result of broadcast_arrays needs to be writeable (for now), to
+    # but the result of broadcast_arrays needs to be writeable, to
     # preserve backwards compatibility
     for results in [broadcast_arrays(original),
                     broadcast_arrays(0, original)]:
         for result in results:
-            assert_equal(result.flags.writeable, True)
+            # This will change to False in a future version
+            if any([s == 0 for s in result.strides]):
+                with assert_warns(FutureWarning):
+                    assert_equal(result.flags.writeable, True)
+                with assert_warns(DeprecationWarning):
+                    result[:] = 0
+                # Warning not emitted, writing to the array resets it
+                assert_equal(result.flags.writeable, True)
+    for results in [broadcast_arrays(original),
+                    broadcast_arrays(0, original)]:
+        for result in results:
+            # resets the warn_on_write DeprecationWarning
+            result.flags.writeable = True
+            result[:] = 0
```

Comment:
Can you try accessing `result.flags.writeable` here too, and prove it raises no warning?

Reply:
ok